### ISSUE TITLE
Disallow Event constructor for IE compatibility (no-event-constructor)

### ISSUE THREAD

**Issue by shanimal** [eslint/eslint#12658]
**Please describe what the rule should do:**
Disallow Event constructor for IE compatibility. Recommend using CustomEvent instead for IE9+ 

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
const evt = new Event("look", {"bubbles":true, "cancelable":false});
const event = new Event('build');
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Many developers struggle to maintain compatibility with IE9+. This rule would prevent this somewhat obscure problem.

**Are you willing to submit a pull request to implement this rule?**
Yes.

**Comment by g-plane** [comment]
Personally, I think this rule should be implemented in a separated plugin, since this is oriented to browser only.

**Comment by kaicataldo** [comment]
I’m in agreement with @g-plane. We have a very high bar for adding new rules to core, and one of the criteria we use to make this determination is if the rule is universal or not.

**Comment by kaicataldo** [comment]
A project like [this one](https://github.com/amilajack/eslint-plugin-compat) would probably be a better fit for this rule.

**Comment by shanimal** [comment]
Sure np. 

`no-restricted-syntax` in my eslint.rc did the trick.

```json
"rules": {
    "no-restricted-syntax": [
        "error",
        {
            "selector": "[type=NewExpression][callee.type = Identifier][callee.name = Event],[type=CallExpression][callee.type = Identifier][callee.name = Event]",
            "message": "Event constructor is not compatible with IE."
        }
    ]
}
```
